### ISSUE TITLE
Some devices are not showing...

### ISSUE THREAD

**Issue by gokhanvarolhotmail** [SamDel/ChromeCast-Desktop-Audio-Streamer#87]
I am running windows 10 on my desktop connected to the router via ethernet. Some of the devices on google home are not showing, some of the deleted devices are still showing. I am connected via ethernet, Upnp is enabled in the router. I have the same problem with my notebook connecting to wireless. The ip address in the setting is right for both.
Both machines are running latest windows 10 pro.
Please advise

[https://1drv.ms/u/s!Apv4g3d243Epib1bjhDJWCIybLE0Rw?e=a9EBVn](url)

**Comment by SamDel** [comment]
Sounds like you've tried a lot of things already. 
To remove the deleted items you can try a 'Reset settings' on the options tab.
Are the bedroom speakers connected to the same router? 
Have you tried rebooting everything (Chromecast devices/router/desktop/notebook) at the same time?

**Comment by FA-Bubba** [comment]
Not sure where to post this...

My house has been in a power outage since Tuesday, but I have a whole-house generator, so everything is powered. The local internet provider is down, so no internet. 

My home network is up & all devices are connected with a 90% or better connection. However DAS shows all Devices with a “Connect Error” notation, so no streaming. Google responds when I make any inquiry “I can’t reach the internet right now...”, so it seems to be working as expected.  I have rebooted all devices and PCs....

Does DAS need an internet connection to work?  If not can you suggest other troubleshooting I could try?

Thank you!

**Comment by SamDel** [comment]
“Connect Error” means something went wrong creating a connection to the device. 
The error is in the log, something like "ex [192.168.0.1] : Connect <technical error string>". Can you post that one?

I would say a internet connection is not neccesary (Only for checking the github site for a new release.). I'll test it later!

**Comment by FA-Bubba** [comment]
This issue is moot now; here is what happened.

My DAS system was not working as described above.  I had tried "Reset Settings" several times, and also unloaded the App and restarted it several times.  EACH time, the Devices all showed the “Connect Error” notation.

About 2pm today, the power company restored service, and my generator went off line.  With the power restoration, the local Comcast/Xfinity service was also restored, which includes internet service.  When that happened, the DAS Device boxes turned green and resumed casting...

This gives the impression that DAS does indeed need the internet signal to work!

Perhaps there is some obscure variable or setting that is set to check for internet, although DAS isn't 'technically using it'...

It occurs to me that there could be a case when someone wants the DAS music, but doesn't want to pay for internet, so they just set up a local area network, without internet -- in that case, and based on my experience, DAS wouldn't work in that environment (but that's just my guess).

**Comment by SamDel** [comment]
Interesting! I'll do a test later and will let you know. It can also be the devices need an internet connection...

**Comment by FA-Bubba** [comment]
Here's the Log...


[Log-2020-08-09-ConnectError.TXT](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/files/5046341/Log-2020-08-09-ConnectError.TXT)

**Comment by FA-Bubba** [comment]
RE:  "Interesting! I'll do a test later and will let you know. It can also be the devices need an internet connection..."

I basically have just 3 types of devices:
 - Google Home Minis -- These MIGHT need an internet connection, but they would respond to my voice inquiries with: "I can’t reach the internet right now..."
 - Google Home Hub -- same as above
 - Google Chromecast Audio (CCA) dongles connected to Philips Stereo speakers.  They do not respond to "Hey Google" at all (not supposed to), and the Speakers do not need internet, BUT CCA might.

Thanks!

**Comment by SamDel** [comment]
Test results:

- When a device is playing and you unplug the (main) internet cable, DAS continues playing.
- When a device is idle and you unplug the (main) internet cable, DAS can start streaming to the device.
- When a device is turned on when no internet is available it refuses connections ("Connect Error") until internet is available.

**Comment by FA-Bubba** [comment]
So... in the 3rd Scenario, is it code in DAS that causes it to refuse the connection?

Can that be changed?

**Comment by SamDel** [comment]
I was not sure so I did another test. 
In the 3rd scenario I tried to connect with the Google Home app. That didn't work either, so I think the device refuses the connection.

**Comment by gokhanvarolhotmail** [comment]
I have 2 Lenovo displays in different rooms with different names. Only one of them is showing in the list.

**Comment by SamDel** [comment]
I don't know why the device doesn't show up, maybe something goes wrong with the discovery.
Can you try to add it in the config file ([see here](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/wiki#configuration))? Just to see if that's working and the device is reachable.

**Comment by FA-Bubba** [comment]
I tried adding the devices to the config file with this line:
<192.168.1.13>,<Family Room Speakers>;<192.168.1.25>,<PlayRoom Speakers>;<192.168.1.11>,<Office Speakers>;<192.168.1.17>,<Guest Room Speakers>;<192.168.1.18>,<Jade's Room Speakers>;<192.168.1.14>,<Basement Speakers>;<192.168.1.19>,<Study Speakers>;<192.168.1.23>,<Dining Room Speakers>;<192.168.1.20>,<Kitchen Hub>;<192.168.1.10>,<Basement Mini>;<192.168.1.15>,<Family Room Mini>;<192.168.1.22>,<Office Mini>;<192.168.1.24>,<Family Room TV>;<192.168.1.26>,<Master Bath Mini>

Then DAS would not start at all...

I removed that line, and DAS started OK (still with some devices missing)...  What is wrong with that line?

**Comment by SamDel** [comment]
It should be like this:

`<add key="IpAddressesDevices" value="192.168.1.13,name1;192.168.1.25,name2;192.168.1.11,name3;192.168.1.17,name4>
`

**Comment by FA-Bubba** [comment]
I just noticed the code line I pasted in was missing a lot -- I guess the Comments App does some filtering if it thinks it sees code...
The code I previously used was:
`<192.168.1.13>,<Family Room Speakers>;<192.168.1.25>,<PlayRoom Speakers>;<192.168.1.11>,<Office Speakers>;<192.168.1.17>,<Guest Room Speakers>;<192.168.1.18>,<Jade's Room Speakers>;<192.168.1.14>,<Basement Speakers>;<192.168.1.19>,<Study Speakers>;<192.168.1.23>,<Dining Room Speakers>;<192.168.1.20>,<Kitchen Hub>;<192.168.1.10>,<Basement Mini>;<192.168.1.15>,<Family Room Mini>;<192.168.1.22>,<Office Mini>;<192.168.1.24>,<Family Room TV>;<192.168.1.26>,<Master Bath Mini>`

For the above string, I used your example on this page:
https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/wiki#configuration

So if I remove the angle brackets around the device addresses & names, and type it as shown your previous comment,  wouldn't the quotes for 'value' need to be closed?  As in:

`<add key="IpAddressesDevices" value="192.168.1.13,name1;192.168.1.25,name2;192.168.1.11,name3;192.168.1.17,name4>">`
-------------------------------------------------------------------------------------------------------------------------------------^

**Comment by FA-Bubba** [comment]
OOPS!  I m eant to type:

`<add key="IpAddressesDevices" value="192.168.1.13,name1;192.168.1.25,name2;192.168.1.11,name3;192.168.1.17,name4">`
-----------------------------------------------------------------------------------------------------------------------------------^

**Comment by SamDel** [comment]
The wiki was wrong, I corrected it after your comment!

You're right, there should be a closing quote.

**Comment by FA-Bubba** [comment]
Add a closing quote, AND a 'slash':
`<add key="IpAddressesDevices" value="192.168.1.13,name1;192.168.1.25,name2;192.168.1.11,name3;192.168.1.17,name4" />`
-------------------------------------------------------------------------------------------------------------------------------------^

After adding the "quote space slash", DAS now loads and casts as expected, BUT it still can't find 3 of my devices (all minis), and it can't find one Speaker Group...

The Minis are all showing on my router with the correct IP Address, but I am unable to ping two of them,  so they may have a connectivity issue.  They all DO respond to the "Hey Google" inquiries, so I'm not sure what the issue is.  The third Mini does respond to a ping, and is visible on my Router, but is not found by DAS...

**Comment by FA-Bubba** [comment]
I'm not sure where to post this, but this is another case of Devices not populating into the DAS Device List...

DAS has had an interesting exception:
One of my speakers (A Google Home Hub) is not populating into the Device list.  I tried "Scan Again...” several times, and “Reset Settings” a few times, still, it would not populate.  I selected the Speaker Group that it belongs to, and the stream started .  When I walked up to the Hub, I noticed it was playing the music stream.  I then checked the Google Home App to see what it showed about the Hub… Oddly, it showed it as off line, and said “Connect to your Wi-Fi to cast”, yet it was playing the DAS music stream.  I asked Google some questions, and it responded with correct answers, so it it indeed connected to Wi-Fi…  I even asked the Hub: "Hey Google, what song is playing now?", and the Hub responded that it was playing "Desktop Audio Streamer"

My router showed the Hub was connected, and with a 99% Experience level...

I cycled the Hub OFF/ON, it reappeared in the DAS Device List, and caused its Speaker Group to go off-line & displayed the 'Orange color'.  Re-selecting the Speaker Group resumed the stream to all Speakers in that Group, and when I checked Google Home App, the Hub was displaying as online.

Long story short, something caused the Google Hub to appear as off-line to both DAS & to Google Home, but the Hub functioned as if it was online, streaming DAS music, and responding to "Hey Google" inquiries.  Cycling its power OFF then ON, corrected its online status.

This is NOT a big issue, and it was easily corrected; I'm just wondering how it received the DAS Stream when DAS "didn't see it"...

**Comment by SamDel** [comment]
When you start streaming to a group DAS connects to the leader of the group and tells it to play the stream. The leader-device then broadcasts the stream to the other devices in the group. I think in this case DAS didn't see the hub-device but DAS did see the leader-device, and the leader-device did see the hub-device. I don't know what happened exactly 😕.
